legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/runtime/Source.java

```diff
@@ -348,15 +434,48 @@ public CoreInstance findElementAt(int line, int column)
             default:
             {
                 Multimap<SourceInformation, CoreInstance> elementsBySourceInfo = elements.groupBy(CoreInstance::getSourceInformation);
-                SourceInformation minSourceInfo = elementsBySourceInfo.keysView().min(Source::compareSourceInformation);
+                SourceInformation minSourceInfo = elementsBySourceInfo.keysView().minBy(sourceInformation ->
+                {
+                    // NOTE: here we consider factor of 10000 is rather safe in order to
+                    // put more weight on line-proximity (over column-proximity)
+                    // unless the line gets really long.
+                    // TODO: find a better way to do this
+                    int LINE_SCORE_FACTOR = 10000;
+                    return Math.abs(sourceInformation.getStartLine() - line) * LINE_SCORE_FACTOR +
+                            Math.abs(sourceInformation.getEndLine() - line) * LINE_SCORE_FACTOR +
+                            Math.abs(sourceInformation.getStartColumn() - column) +
+                            Math.abs(sourceInformation.getEndColumn() - column);
```

Comment:
If you're going to be multiplying, you'll need to worry about integer overflow. It's better if you can find a way to do the comparison without multiplying.

Also, why bother with Math.abs? You know that line is between startLine and endLine and column is between startColumn and endColumn.